madprops/serializers.py

```diff
@@ -0,0 +1,127 @@
+from django.utils.functional import cached_property
+from rest_framework.serializers import (
+    ModelSerializer, ModelSerializerOptions, RelationsList)
+
+
+class PropertiesSerializerOptions(ModelSerializerOptions):
+    """Meta class options for PropertiesSerializer"""
+    def __init__(self, meta):
+        super(PropertiesSerializerOptions, self).__init__(meta)
+        self.parent_obj_name = getattr(meta, 'parent_obj_name', None)
+        self.read_only_props = getattr(meta, 'read_only_props', [])
+        self.exclude = ('id',)
+
+
+class NestedPropertiesSerializerOptions(PropertiesSerializerOptions):
+    """Meta class options for NestedPropertiesSerializer"""
+    def __init__(self, meta):
+        super(NestedPropertiesSerializerOptions, self).__init__(meta)
+        self.exclude = ('id', self.parent_obj_name)
+
+
+class PropertiesSerializer(ModelSerializer):
```

Comment:
IMO your suggested name is a bit generic, PropertiesSerializer might not be good - but it better describes what is't used for.  Let's hear opinions of other ppl.